pkg/kn/config/config.go

```diff
@@ -0,0 +1,259 @@
+func initDefaults() *defaultConfig {
+	return &defaultConfig{
+		configFile:          defaultConfigLocation("config.yaml"),
+		pluginsDir:          defaultConfigLocation("plugins"),
+		lookupPluginsInPath: false,
+	}
+}
+
+func defaultConfigLocation(subpath string) string {
+	var base string
+	if runtime.GOOS == "windows" {
+		base = defaultConfigDirWindows()
+	} else {
+		base = defaultConfigDirUnix()
+	}
+	return filepath.Join(base, subpath)
+}
+
+func defaultConfigDirUnix() string {
+	home, err := homedir.Dir()
+	if err != nil {
+		home = "~"
+	}
+
+	// Check the deprecated path first and fallback to it, add warning to error message
+	if configHome := filepath.Join(home, ".kn"); dirExists(configHome) {
+		migrationPath := filepath.Join(home, ".config", "kn")
+		fmt.Fprintf(os.Stderr, "WARNING: deprecated kn config directory '%s' detected. Please move your configuration to '%s'\n", configHome, migrationPath)
+		return configHome
+	}
+
+	// Respect XDG_CONFIG_HOME if set
+	if xdgHome := os.Getenv("XDG_CONFIG_HOME"); xdgHome != "" {
+		return filepath.Join(xdgHome, "kn")
+	}
+	// Fallback to XDG default for both Linux and macOS
+	// ~/.config/kn
+	return filepath.Join(home, ".config", "kn")
+}
+
+func defaultConfigDirWindows() string {
+	home, err := homedir.Dir()
+	if err != nil {
+		// Check the deprecated path first and fallback to it, add warning to error message
+		if configHome := filepath.Join(home, ".kn"); dirExists(configHome) {
+			migrationPath := filepath.Join(os.Getenv("APPDATA"), "kn")
+			fmt.Fprintf(os.Stderr, "WARNING: deprecated kn config directory '%s' detected. Please move your configuration to '%s'\n", configHome, migrationPath)
+			return configHome
+		}
+	}
+
+	return filepath.Join(os.Getenv("APPDATA"), "kn")
+}
+
+func dirExists(path string) bool {
+	if _, err := os.Stat(path); !os.IsNotExist(err) {
+		return true
+	}
+	return false
+}
+
+// parse sink mappings and store them in the global configuration
+func parseSinkMappings() error {
+	// Parse sink configuration
+	key := ""
+	if viper.IsSet(keySinkMappings) {
+		key = keySinkMappings
+	}
+	if key == "" && viper.IsSet(legacyKeySinkMappings) {
+		key = legacyKeySinkMappings
+	}
+
+	if key != "" {
+		err := viper.UnmarshalKey(key, &globalConfig.sinkMappings)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("error while parsing sink mappings in configuration file %s",
+				viper.ConfigFileUsed()))
+		}
+	}
+	return nil
+}
+
+// Prepare the default config file for the usage message
+func defaultConfigFileForUsageMessage() string {
+	// We can't use the real default value here as this will appear then
+	// in the docs resolved (like '/Users/roland/.config/kn/config.yaml' when building on my machine)
+	if runtime.GOOS == "windows" {
+		return "%APPDATA%\\kn"
+	}
+	homeDir, err := homedir.Dir()
+	if err == nil {
+		return strings.Replace(bootstrapDefaults.configFile, homeDir, "~", -1)
+	} else {
+		return "~/.config/kn/config.yaml"
+	}
```

Comment:
```suggestion
	// We can't use the real default value here as this will appear then
	// in the docs resolved (like '/Users/roland/.config/kn/config.yaml' when building on my machine)
	if runtime.GOOS == "windows" {
		return "%APPDATA%\\kn\\config.yaml"
	}
      return "~/.config/kn/config.yaml"
```
since this is just for help message, we dont need to resolve the home directory for non-Windows platforms ?
Also, suggested to add `\\config.yaml` for Windows.

Reply:
the problem if it is reslolved for the usage message, the doc generation will pick up my home directory literally (`/Users/roland`) and put it into the docs, which is not as it should be (and also leads to build failures because of the diff)

> Also, suggested to add `\\config.yaml` for Windows.

Isn't it added ?

Reply:
> the problem if it is reslolved for the usage message, the doc generation will pick up my home directory

correct, I am suggesting to not resolve the homedir at all (its anyway replaced by `~` for non-Windows platforms, we can directly use `~`)

Reply:
> return "~/.config/kn/config.yaml"

I'm still would use a replacement as this can be picked up from the config file, too. But well, for the usage message I think your suggestion is good enough. Thanks !